apps/emqx/src/emqx_frame.erl

```diff
@@ -448,9 +486,21 @@ parse_utf8_string(Bin, false) ->
 parse_utf8_string(Bin, true) ->
     parse_utf8_string(Bin).
 
+parse_utf8_string(Bin)
+  when 2 > byte_size(Bin) ->
+    ?PARSE_ERR(utf8_length_error);
+parse_utf8_string(<<Len:16/big, Rest/binary>>)
+  when Len > byte_size(Rest) ->
+    ?PARSE_ERR(utf8_length_error);
```

Comment:
move this two clauses down to after the normal clause.

and provide more info in the exception.
e.g.
`?PARSE_ERR(malformed_utf8_string_length)`
and
`?PARSE_ERR(#{hint => malformed_utf8_string, parsed_length => Len, remaining_bytes => byte_size(Rest)})`